File: java_service/src/main/java/com/hlbank/processing/fee/FeeService.java
package com.hlbank.processing.fee;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

@Component
public class FeeService {
    private final Logger logger = LoggerFactory.getLogger(this.getClass());
    private static final int FEE_MIN = 100;
    private static final double FEE_PERCENT_100k = 0.025;
    private static final double FEE_PERCENT_10k = 0.05;

    public Integer calculateFee(Integer amount) {
        logger.info("calculating fee for amount " + amount);
        Integer fee = FEE_MIN;

        if (amount == null)
            return 0;
        else if (amount > 100000) fee = (int)Math.floor(amount * FEE_PERCENT_100k);
        else if (amount > 10000) fee = (int)Math.floor(amount * FEE_PERCENT_10k);

        return fee;
    }
}

File: nodejs_service/src/fee/fee.service.ts
import { Injectable } from '@nestjs/common';

export const FEE_MIN = 100;
export const FEE_PERCENT_100k = 0.025;
export const FEE_PERCENT_10k = 0.05;

@Injectable()
export class FeeService {
    calculateFee(amount: number | null): number | null {
        let fee = FEE_MIN;

        if (amount === null)
            return null;
        else if (amount > 100000) fee = Math.floor(amount * FEE_PERCENT_100k);
        else if (amount > 10000) fee = Math.floor(amount * FEE_PERCENT_10k);

        return fee;
    }
}
